# Exploratory Data Analysis

In [1]:
import pandas as pd

In [15]:
df_x = pd.read_csv('data/data_x.csv', index_col='id')
df_y = pd.read_csv('data/data_y.csv', index_col='id')
df = pd.concat([df_y, df_x], axis=1)

In [16]:
df

status_group  amount_tsh date_recorded           funder  gps_height  \
id                                                                             
69572      functional      6000.0    2011-03-14            Roman        1390   
8776       functional         0.0    2013-03-06          Grumeti        1399   
34310      functional        25.0    2013-02-25     Lottery Club         686   
67743  non functional         0.0    2013-01-28           Unicef         263   
19728      functional         0.0    2011-07-13      Action In A           0   
...               ...         ...           ...              ...         ...   
60739      functional        10.0    2013-05-03  Germany Republi        1210   
27263      functional      4700.0    2011-05-07      Cefa-njombe        1212   
37057      functional         0.0    2011-04-11              NaN           0   
31282      functional         0.0    2011-03-08            Malec           0   
26348      functional         0.0    2011-03-23       World Bank         191   

          installer  longitude   latitude              wpt_name  num_private  \
id                                                                             
69572         Roman  34.938093  -9.856322                  none            0   
8776        GRUMETI  34.698766  -2.147466              Zahanati            0   
34310  World vision  37.460664  -3.821329           Kwa Mahundi            0   
67743        UNICEF  38.486161 -11.155298  Zahanati Ya Nanyumbu            0   
19728       Artisan  31.130847  -1.825359               Shuleni            0   
...             ...        ...        ...                   ...          ...   
60739           CES  37.169807  -3.253847   Area Three Namba 27            0   
27263          Cefa  35.249991  -9.070629     Kwa Yahona Kuvala            0   
37057           NaN  34.017087  -8.750434               Mashine            0   
31282          Musa  35.861315  -6.378573                Mshoro            0   
26348         World  38.104048  -6.747464       Kwa Mzee Lugawa            0   

       ... payment_type water_quality quality_group      quantity  \
id     ...                                                          
69572  ...     annually          soft          good        enough   
8776   ...    never pay          soft          good  insufficient   
34310  ...   per bucket          soft          good        enough   
67743  ...    never pay          soft          good           dry   
19728  ...    never pay          soft          good      seasonal   
...    ...          ...           ...           ...           ...   
60739  ...   per bucket          soft          good        enough   
27263  ...     annually          soft          good        enough   
37057  ...      monthly      fluoride      fluoride        enough   
31282  ...    never pay          soft          good  insufficient   
26348  ...   on failure         salty         salty        enough   

       quantity_group                source           source_type  \
id                                                                  
69572          enough                spring                spring   
8776     insufficient  rainwater harvesting  rainwater harvesting   
34310          enough                   dam                   dam   
67743             dry           machine dbh              borehole   
19728        seasonal  rainwater harvesting  rainwater harvesting   
...               ...                   ...                   ...   
60739          enough                spring                spring   
27263          enough                 river            river/lake   
37057          enough           machine dbh              borehole   
31282    insufficient          shallow well          shallow well   
26348          enough          shallow well          shallow well   

       source_class              waterpoint_type waterpoint_type_group  
id                                                        

59400 data points, 39 dependent variables

In [19]:
for col in df.columns:
    print(df[col].value_counts(normalize=True), '\n')

functional                 0.543081
non functional             0.384242
functional needs repair    0.072677
Name: status_group, dtype: float64 

0.0         0.700993
500.0       0.052222
50.0        0.041616
1000.0      0.025051
20.0        0.024630
              ...   
8500.0      0.000017
6300.0      0.000017
220.0       0.000017
138000.0    0.000017
12.0        0.000017
Name: amount_tsh, Length: 98, dtype: float64 

2011-03-15    0.009630
2011-03-17    0.009394
2013-02-03    0.009192
2011-03-14    0.008754
2011-03-16    0.008636
                ...   
2011-09-14    0.000017
2004-04-05    0.000017
2011-09-16    0.000017
2011-09-27    0.000017
2011-08-31    0.000017
Name: date_recorded, Length: 356, dtype: float64 

Government Of Tanzania    0.162898
Danida                    0.055841
Hesawa                    0.039487
Rwssp                     0.024639
World Bank                0.024191
                            ...   
Rafael Michael            0.000018
Runda                     0.

- class imbalance in status_group (dependent variable): 54%-38%-7% split
- amount_tsh: over 70% 0 - an actual value of 0 would indicate no water, so this feature is likely very incorrect, and mostly useless
- date_recorded: over a range of dates from 2002 to 2013
- gps_height: over a third are at exactly zero, with another 2.5% below zero, I'm not sure what to make of this, but the data will likely not be used
- longitude and latitude: the points with 0 longitude match exactly with -2e-8 latitude, neither of which are in Tanzania, but these only take up 3% of the data
- num_private: no description given for this data, will need to dig deeper
- population: 36% of the data at 0 means this isn't likely factual information, but it seems useful, so I will need to research more before dismissing
- public_meeting: no information given about this feature
- construction_year: the 35% of values with a 0 are obviously wrong
- payment and payment_type: these features seemt o be exact copies of each other
- water_quality and quality_group: near copies of each other, water_quality is a bit more precise
- quantity and quantity_group: exact copies of each other
- source, source_type, and source_class: near copies of each other, source is a bit more precise
- waterpoint_type and waterpoint_type_group: near copies of each other, waterpoint_type is a bit more precise